src/main/java/seedu/address/logic/commands/FindDeliveryCommand.java

```diff
@@ -0,0 +1,99 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATETIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DELIVERY;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_PRODUCT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_STATUS;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.delivery.Delivery;
+import seedu.address.model.delivery.DeliverySupplierPredicate;
+import seedu.address.model.delivery.SupplierIndex;
+import seedu.address.model.person.Person;
+
+/**
+ * Finds and lists all deliveries in the address book that match the specified criteria.
+ */
+public class FindDeliveryCommand extends Command {
+
+    public static final String COMMAND_WORD = "find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + " " + PREFIX_DELIVERY
+            + " : Finds all deliveries based on the provided filters.\n"
+            + "Parameters: "
+            + "[" + PREFIX_DATETIME + "DELIVERY_DATE] "
```

Comment:
i think have to change these 3 and remove the [ ] cause this indicates that multiple prefixes are allowed according to the user guide